airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/IntegrationRunner.java

```diff
@@ -147,9 +145,20 @@ private void runInternal(final IntegrationConfig parsed) throws Exception {
           final JsonNode config = parseConfig(parsed.getConfigPath());
           validateConfig(integration.spec().getConnectionSpecification(), config, "WRITE");
           final ConfiguredAirbyteCatalog catalog = parseConfig(parsed.getCatalogPath(), ConfiguredAirbyteCatalog.class);
-          try (final AirbyteMessageConsumer consumer = destination.getConsumer(config, catalog, outputRecordCollector)) {
-            runConsumer(consumer);
-          }
+
+          final Procedure consumeWriteStreamCallable = () -> {
+            try (final SerializedAirbyteMessageConsumer consumer = destination.getSerializedMessageConsumer(config, catalog, outputRecordCollector)) {
+              consumeWriteStream(consumer);
+            }
+          };
+
+          watchForOrphanThreads(
```

Comment:
for context, we moved this because on master it is getting called before the `close` is called on the destination. this doesn't make sense, because it is checking if the destination cleaned itself up properly. that's a ridiculous thing to change before you let the destination call `close`. now this only gets called _after_ the destination calls `close`